staging/src/slime.io/slime/modules/lazyload/controllers/service_cache.go

```diff
@@ -115,8 +96,106 @@ func newSvcCache(clientSet *kubernetes.Clientset) (*NsSvcCache, *LabelSvcCache,
 			}
 			labelSvcCache.Unlock()
 
+			// add eventSvc ports to portProtocolCache again
+			if ns != r.env.Config.Global.IstioNamespace {
+				portProtocolCache.Lock()
+				for _, port := range service.Spec.Ports {
+					p := port.Port
+					portProtos := portProtocolCache.Data[p]
+					if portProtos == nil {
+						portProtos = make(map[Protocol]uint)
+						portProtocolCache.Data[p] = portProtos
+					}
+					proto := getProtocol(port)
+					portProtos[proto]++
+				}
+				portProtocolCache.Unlock()
+			}
+
 		}
 	}()
 
-	return nsSvcCache, labelSvcCache, nil
+	needUpdate, successUpdate := false, true
+
+	if r.cfg.AutoPort {
+		log.Infof("Lazyload port auto management is running")
+		go func() {
+			// polling request
+			pollTicker := time.NewTicker(10 * time.Second)
+			// init and retry request
+			retryTicker := time.NewTicker(1 * time.Second)
+			for {
+				select {
+				case <-pollTicker.C:
+				case <-retryTicker.C:
+					retryTicker.C = nil
```

Comment:
a little weird to set-nil the `ticker.C`. it's more considered as read-only thing.

write it like this:

```go
var retryCh <- chan time.Time

// ...

case <- retryCh:
  retryCh = nil

// ...
retryCh = time.After(time.Second)
```